### PR TITLE
[z3] Update to 4.8.14

### PR DESCRIPTION
**Pull request by Shatur**
**Describe the pull request**

- #### What does your PR fix?  
Updates z3 to 4.8.14

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
 All, no

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
 Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
 Yes

### CODE REVIEW

**Comment by Shatur** [comment]
CI error seems unrelated?

**Comment by JonLiu1993** [comment]
@Shatur ,Thanks for your pr, Could you refer to @chausner 's [pr ](https://github.com/microsoft/vcpkg/pull/22708)to modernize the functions in portfile.cmake?

File: ports/z3/portfile.cmake [hunk 1]
```diff
@@ -17,15 +17,14 @@ if (VCPKG_LIBRARY_LINKAGE STREQUAL "static")
...
+vcpkg_cmake_configure(
+  SOURCE_PATH "${SOURCE_PATH}"
   OPTIONS
     ${BUILD_STATIC}
 )
 
-vcpkg_install_cmake()
-vcpkg_fixup_cmake_targets(CONFIG_PATH lib/cmake/z3 TARGET_PATH share/Z3)
+vcpkg_cmake_install()
+vcpkg_cmake_config_fixup(PACKAGE_NAME Z3 CONFIG_PATH lib/cmake/z3)
```
Comment (JonLiu1993) [inline ports/z3/portfile.cmake:27]:
```suggestion
vcpkg_cmake_config_fixup(PACKAGE_NAME z3 CONFIG_PATH lib/cmake/z3)
```

Comment (JonLiu1993) [inline ports/z3/portfile.cmake:27]:
```suggestion
vcpkg_cmake_config_fixup(PACKAGE_NAME z3 CONFIG_PATH lib/cmake/z3)
```

**Review by JonLiu1993 — APPROVED** [review]

**Comment by Shatur** [comment]
> @Shatur, Looks the ci test failed , this is error log ,please take a look:

Yep, can reproduce it locally. The error appears after CMake functions modernization:

```
CMake Error at C:/ProgramData/scoop/apps/cmake/3.22.0/share/cmake-3.22/Modules/ExternalProject.cmake:1738 (file):
  file attempted to create a directory:
  C:/ProgramData/vcpkg/buildtrees/z3/src/z3-4.8.14-c59fdb3a44.clean/examples/c
  into a source directory.
Call Stack (most recent call first):
  C:/ProgramData/scoop/apps/cmake/3.22.0/share/cmake-3.22/Modules/ExternalProject.cmake:3652 (_ep_set_directories)
  examples/CMakeLists.txt:37 (ExternalProject_Add)


-- Configuring incomplete, errors occurred!
```

Why it could happen?

File: ports/z3/portfile.cmake [hunk 2]
```diff
@@ -17,15 +17,14 @@ if (VCPKG_LIBRARY_LINKAGE STREQUAL "static")
   set(BUILD_STATIC "-DZ3_BUILD_LIBZ3_SHARED=OFF")
 endif()
 
-vcpkg_configure_cmake(
-  SOURCE_PATH ${SOURCE_PATH}
-  PREFER_NINJA
+vcpkg_cmake_configure(
+  SOURCE_PATH "${SOURCE_PATH}"
```
Comment (JonLiu1993) [inline ports/z3/portfile.cmake:21]:
```suggestion
vcpkg_configure_cmake(
  SOURCE_PATH "${SOURCE_PATH}"
  PREFER_NINJA
```

Comment (JonLiu1993) [reply ports/z3/portfile.cmake:21]: @Shatur , It looks like this is a bug in vcpkg_cmake_configure, refer to https://github.com/microsoft/vcpkg/pull/22124 , You can restore this function back

**Comment by JonLiu1993** [comment]
> > @Shatur, Looks the ci test failed , this is error log ,please take a look:
> 
> Yep, can reproduce it locally. The error appears after CMake functions modernization (it worked before it):
> 
> ```
> CMake Error at D:/downloads/tools/cmake-3.21.1-windows/cmake-3.21.1-windows-i386/share/cmake-3.21/Modules/ExternalProject.cmake:1738 (file):
>   file attempted to create a directory:
>   D:/buildtrees/z3/src/z3-4.8.14-c59fdb3a44.clean/examples/c into a source
>   directory.
> Call Stack (most recent call first):
>   D:/downloads/tools/cmake-3.21.1-windows/cmake-3.21.1-windows-i386/share/cmake-3.21/Modules/ExternalProject.cmake:3652 (_ep_set_directories)
>   examples/CMakeLists.txt:37 (ExternalProject_Add)
> 
> 
> -- Configuring incomplete, errors occurred!
> ```
> 
> Why it could happen? I'm unfamiliar with vcpkg-specific CMake functions. I can revert the modernization, it works without it.

@Shatur , It looks like this is a bug in vcpkg_cmake_configure, refer to https://github.com/microsoft/vcpkg/pull/22124 , You can restore this function back

**Comment by Shatur** [comment]
> @Shatur , It looks like this is a bug in vcpkg_cmake_configure, refer to #22124 , You can restore this function back

It looks like I can't revert only this function. I have the following error after this:

```
CMake Error at scripts/cmake/vcpkg_configure_cmake.cmake:110 (message):
  The z3 port already depends on vcpkg-cmake; using both vcpkg-cmake and
  vcpkg_configure_cmake in the same port is unsupported.
Call Stack (most recent call first):
  ports/z3/portfile.cmake:20 (vcpkg_configure_cmake)
  scripts/ports.cmake:142 (include)


Error: Building package z3:x64-windows failed with: BUILD_FAILED
```

**Review by JonLiu1993 — APPROVED** [review]

**Comment by Shatur** [comment]
@JonLiu1993 is there anything else preventing it to be merged?

**Comment by ras0219-msft** [comment]
The changes LGTM, but unfortunately it looks like it conflicts with master at the moment. I tried to push a merge but I believe you have pushes disabled for upstream maintainers.

When merging, I recommend discarding your changes to the version files in `versions/` and then regenerating them with `vcpkg x-add-version z3` afterwards.

**Comment by Shatur** [comment]
> The changes LGTM, but unfortunately it looks like it conflicts with master at the moment.

Thanks, missed it!

>  I tried to push a merge but I believe you have pushes disabled for upstream maintainers.

Hm... For some reason I don't have a checkbox for it in this PR, strange.

> When merging, I recommend discarding your changes to the version files in versions/ and then regenerating them with vcpkg x-add-version z3 afterwards.

Sure, updated to the latest master and regenerated versions/ update.

**Review by JonLiu1993 — APPROVED** [review]

**Pull request merged**